go.mod

```diff
@@ -12,7 +12,7 @@ require (
 	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice/v4 v4.2.0
 	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v4 v4.1.0
 	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources v1.1.1
-	github.com/Azure/azure-service-operator/v2 v2.2.0
+	github.com/Azure/azure-service-operator/v2 v2.3.0
```

Comment:
@jackfrancis In case ASO v2.4 releases after I go OOF next week, this PR should serve as a good model for how to update that. The only additional change required this time is a new `replace` here in go.mod:
```go.mod
// This accounts for a breaking change between CAPI's version and ASO's version,
// reverting to CAPI's version.
replace k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20230717233707-2695361300d9
```

Reply:
There might be a little more finagling required here if we want to update ASO to 2.4 before bumping CAPI to 1.6: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4182#issuecomment-1802649734